src/components/UTIL_FormField.component

```diff
@@ -0,0 +1,113 @@
+    <apex:attribute name="sObj" type="sObject" description="The sObject the field belongs to." required="true"/>
+    <apex:attribute name="sObjType" type="String" description="The sObject type." required="true"/>
+    <apex:attribute name="onchange" type="String"  default="" description="On change event."/>
+    <apex:attribute name="required" type="Boolean" default="false" description="When true marks the field as required, defaults to false." />
+    <apex:attribute name="styleClass" type="String" default="" description="The style to apply to the outer form element." />
+
+    <!-- Action Support assumes onchange event and calling page has an action status component on the page -->
+    <apex:attribute name="actSup" type="Boolean" default="false" description="When true the input field will render action support, defaults to false." />
+    <apex:attribute name="actSupAction" type="ApexPages.Action" default="" description="Action for the onchange action support to fire." />
+    <apex:attribute name="actSupReRender" type="String" default="" description="Component id on the main page to rerender after onchange event." />
+    <apex:attribute name="actSupStatus" type="String" default="loading" description="The id of the action status on the main page to display when fired, defaults to loading." />
+    <apex:attribute name="actSupImmediate" type="Boolean" default="false" description="Sets immediate attribute on the action status, defaults to false" />
+
+    <!-- Dependent picklists and multipicklists require additional css -->
+    <apex:variable var="dependent" value="{!$ObjectType[sObjType].fields[field].dependentPicklist}"/>
+    <!-- NOTE: When using a variable for determining readonly or field type in combination with conditionally rendering output panels, some field values do not save -->
+    <apex:variable var="ftype" value="{!$ObjectType[sObjType].fields[field].type}"/>
+    <!-- The fixup style class to apply to dependent and date fields -->
+    <apex:variable var="fixup"
+        value="{!IF(dependent,
+                    IF(ftype=='picklist', ' dependentPicklistFixup', ' dependentMultiPicklistFixup'),
+                    IF(CONTAINS(ftype,'date'),' dateFixup', ''))}"/>
+    <!-- The field label to be displayed -->
+    <apex:variable var="label" value="{!$ObjectType[sObjType].fields[field].label}"/>
+
+    <!-- Read only fields will be displayed as outputfields with a bottom divider per slds standards, checkboxes will render as disabled input checkboxes -->
+    <apex:outputPanel layout="none" id="divOutput"
+        rendered="{!((!$ObjectType[sObjType].fields[field].updateable && sObj['Id']!='')
+                    || !$ObjectType[sObjType].fields[field].createable) && $ObjectType[sObjType].fields[field].accessible}">
+        <div class="{!'slds-form-element ' + styleClass}">
+            <apex:outputLabel for="outputX" styleClass="slds-form-element__label" rendered="{!ftype!='boolean'}">
+                {!label}
```

Comment:
should this also be displaying the red * for required fields?  did any of the uses you are replacing with this in V4S already display a required indicator?